core/connection.js

```diff
@@ -619,3 +621,32 @@ Blockly.Connection.prototype.setShadowDom = function(shadow) {
 Blockly.Connection.prototype.getShadowDom = function() {
   return this.shadowDom_;
 };
+
+/**
+ * @return {string} A human-readable description of this Connection.
+ */
+Blockly.Connection.prototype.describe = function() {
```

Comment:
Again, this would make a better toString function.  Then you could use:
  msg = 'connected to ' + connection;
instead of:
  msg = 'connected to ' + connection.describe();